Sources/_StringProcessing/ByteCodeGen.swift

```diff
@@ -62,7 +62,11 @@ fileprivate extension Compiler.ByteCodeGen {
       emitCharacter(c)
 
     case let .scalar(s):
-      emitScalar(s)
+      if options.semanticLevel == .graphemeCluster {
+        emitCharacter(Character(s))
+      } else {
+        emitMatchScalar(s)
+      }
```

Comment:
This should fail tests, and if it's not we may need to write the tests and XFAIL them. E.g. `/a\u{301}/`, IIUC, as written would try to make a proper Character from the combining scalar.